src/guide/reactivity-computed-watchers.md

```diff
@@ -211,6 +212,79 @@ firstName.value = "John"; // logs: ["John",""] ["", ""]
 lastName.value = "Smith"; // logs: ["John", "Smith"] ["John", ""]
 ```
 
+### Watching Reactive Objects
+
+Using a watcher to compare values of an Array or Object that are reactive requires that it has a copy made of just the values.
+
+```js
+const numbers = reactive([1, 2, 3, 4])
+
+watch(
+  () => [...numbers],
+  (numbers, prevNumbers) => {
+    console.log(numbers, prevNumbers);
+  })
+
+numbers.push(5) // logs: [1,2,3,4,5] [1,2,3,4]
+```
+
+Attempting to check for changes of properties in a deeply nested Object or Array will still require the `deep` option to be true, if watching an Object or Array.
+
+```js
+const state = reactive([
+  { attributes: { name: "" }, id: 1 },
+  { attributes: { name: "" }, id: 2 },
+]);
+
+watch(
+  () => state.map((e) => e.attributes),
+  (attrs, prevAttrs) => {
+    console.log(
+      'not deep ', 
+      attrs.map((e) => e.name), 
+      prevAttrs.map((e) => e.name)
+    );
+  }
+);
+
+watch(
+  () => state.map((e) => e.attributes),
+  (attrs, prevAttrs) => {
+    console.log(
+      'deep ', 
+      attrs.map((e) => e.name), 
+      prevAttrs.map((e) => e.name)
+    );
+  },
+  { deep: true }
+)
+
+state[0].attributes.name = "Alex"; // Logs: "deep " ["Alex", ""] ["Alex", ""]
+```
+
+However, watching a reactive Object or Array will always return a reference to the current value of that Object. To fully watch deeply nested Objects and Arrays, a deep copy of values may be required. This can be achieved with a utility such as [lodash.cloneDeep](https://lodash.com/docs/4.17.15#cloneDeep)
```

Comment:
```suggestion
However, watching a reactive object or array will always return a reference to the current value of that object. To fully watch deeply nested objects and arrays, a deep copy of values may be required. This can be achieved with a utility such as [lodash.cloneDeep](https://lodash.com/docs/4.17.15#cloneDeep)
```